src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,388 @@
+-- structure back along `double_centralizer.prod_mk`.
+instance : add_comm_group 𝓜(𝕜, A) :=
+{ add := (+),
+  add_assoc := λ a b c, by {ext; exact add_assoc _ _ _},
+  zero := 0,
+  zero_add := λ a, by {ext; exact zero_add _},
+  add_zero := λ a, by {ext; exact add_zero _},
+  neg := λ x, -x,
+  sub := λ x y,  x - y,
+  sub_eq_add_neg := λ a b, by {ext; exact sub_eq_add_neg _ _},
+  add_left_neg := λ a, by {ext; exact add_left_neg _},
+  add_comm := λ a b, by {ext; exact add_comm _ _}, }
+
+/-- The canonical map `double_centralizer.prod_mk` as an additive group homomorphism. -/
+def add_group_hom_prod_mk : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A) :=
+{ to_fun := prod_mk 𝕜 A,
+  map_zero' := rfl,
+  map_add' := λ x y, rfl }
+
+/-- The module structure is inherited as the pullback under the additive group monomoprhism
+`double_centralizer.prod_mk : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)` -/
+instance : module 𝕜 𝓜(𝕜, A) :=
+function.injective.module 𝕜 add_group_hom_prod_mk injective_prod_mk (λ x y, rfl)
+
+/-- The normed group structure is inherited as the pullback under the additive group monomoprhism
+`double_centralizer.prod_mk : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)` -/
+instance : normed_add_comm_group 𝓜(𝕜, A) :=
+normed_add_comm_group.induced add_group_hom_prod_mk injective_prod_mk
+
+@[simp] lemma norm_eq (a : 𝓜(𝕜, A)) : ∥a∥ = max (∥a.left∥) (∥a.right∥) := rfl
+
+instance : normed_space 𝕜 𝓜(𝕜, A) :=
+{ norm_smul_le := λ k a, show max (∥k • a.left∥) (∥k • a.right∥) ≤ ∥k∥ * max (∥a.left∥) (∥a.right∥),
+    by simp only [mul_max_of_nonneg _ _ (norm_nonneg k), norm_smul],
+  .. double_centralizer.module }
+
+lemma uniform_embedding_prod_mk : uniform_embedding (prod_mk 𝕜 A) :=
+uniform_embedding_comap injective_prod_mk
+
+instance [complete_space A] : complete_space 𝓜(𝕜, A) :=
+begin
+  rw complete_space_iff_is_complete_range uniform_embedding_prod_mk.to_uniform_inducing,
+  apply is_closed.is_complete,
+  simp only [range_prod_mk, set.set_of_forall],
+  refine is_closed_Inter (λ x, is_closed_Inter $ λ y, is_closed_eq _ _),
+  { exact ((continuous_mul_right y).comp (continuous_linear_map.apply 𝕜 A x).continuous).comp
+      continuous_snd },
+  { exact ((continuous_mul_left x).comp (continuous_linear_map.apply 𝕜 A y).continuous).comp
+      continuous_fst }
+end
+
+/-!
+### Multiplicative structure
+-/
+
+instance : has_one 𝓜(𝕜, A) :=
+{ one :=
+  { left := 1,
+    right := 1,
+    central := λ x y, rfl } }
+
+instance : has_mul 𝓜(𝕜, A) :=
+{ mul := λ a b,
+  { left := a.left.comp b.left,
+    right := b.right.comp a.right,
+    central := λ x y, by simp only [continuous_linear_map.coe_comp', function.comp_app, central]}}
+
+@[simp] lemma one_left : (1 : 𝓜(𝕜, A)).left = 1 := rfl
+@[simp] lemma one_right : (1 : 𝓜(𝕜, A)).right = 1 := rfl
+@[simp] lemma mul_left (a b : 𝓜(𝕜, A)) : (a * b).left = a.left * b.left := rfl
+@[simp] lemma mul_right (a b : 𝓜(𝕜, A)) : (a * b).right = b.right * a.right := rfl
+
+instance : ring 𝓜(𝕜, A) :=
```

Comment:
Can you add a lemma about `(↑n : 𝓜(𝕜, A)).left` etc, and similar for `int_cast` and `pow`?

Again, if you do the extra work when building this instance then you don't have to do any work to prove these lemmas.